kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -450,6 +534,39 @@ namespace Kratos
 		}
 	}
 
+	template<std::size_t TDim>
+	void CalculateDiscontinuousDistanceToSkinProcess<TDim>::SetElementalExtrapolatedEdgeDistancesValues(
+        Element& rElement,
+        const array_1d<double, (TDim == 2) ? 3 : 6>& rCutExtraEdgesRatioVector)
+	{
+		// Get reference to ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES and resize if necessary
+		constexpr std::size_t n_edges = (TDim == 2) ? 3 : 6;
+		Vector& r_extra_edge_distances = rElement.GetValue(ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES);
+		if (r_extra_edge_distances.size() != n_edges) {
+			r_extra_edge_distances.resize(n_edges, false);
+		}
+
+		// Save the obtained edge distances in the elemental variable
+		for (std::size_t i = 0; i < n_edges; ++i) {
+			r_extra_edge_distances[i] = rCutExtraEdgesRatioVector[i];
+		}
+	}
```

Comment:
I think this function can be avoided. Just do `rElement.GetValue(ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES) = the_new_values_array`.

Reply:
You think the resizing is not necessary?
Why did you create SetElementalEdgeDistancesValues()?
I thought it makes CalculateElementalAndEdgeDistances() more clear as well.

Reply:
Actually, these values should be already initialized in order to have a threadsafe access since we are working with the non-historical database. This is most probably the reason why I created the `SetElementalEdgeDistancesValues`.

Reply:
They are initialized inside `Initialize()` in a "block_for_each", without being resized, if `block_for_each` doesn't resize.

Reply:
If a `SetValue` is done in the `Initialize` it should be fine.

Reply:
I'm a bit confused, so I will delete the `Set...` methods and just assign the correct values after `GetValue(..)` without resizing? For all three: `SetElementalEdgeDistancesValues`, `SetElementalExtrapolatedEdgeDistancesValues`, `SetElementalDistancesWithExtrapolatedValues`?

Reply:
Actually `SetElementalDistancesWithExtrapolatedValue` would not needed the least, if I would treat it the same way as `ELEMENTAL_DISTANCES`, because the `Vector&` is already there anyway (see conversation about Vector) and then there would be no local variable `Vector`, which we don't want to have at all, do we?